lib/src/components/NxModal/NxModal.tsx

```diff
@@ -4,22 +4,42 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import React, {FunctionComponent} from 'react';
+import React, {FunctionComponent, useEffect} from 'react';
 import classnames from 'classnames';
 
-import {Props, propTypes} from './types';
+import {CloseHandler, Props, propTypes} from './types';
 
 import './NxModal.scss';
 
-const NxModal: FunctionComponent<Props> = ({className, ...attrs}) => {
+const currentModalCloseHandlers: CloseHandler[] = [];
 
+const NxModal: FunctionComponent<Props> = ({className, onClose, ...attrs}) => {
   const modalClasses = classnames('nx-modal', className);
+
+  const modalCloseListener = ({ key }: { key: string}) => {
```

Comment:
The parameter type would be `KeyboardEvent` (the one from react, not the native one)

Reply:
> The parameter type would be `KeyboardEvent` (the one from react, not the native one)

This change would prevent the function being added as a listener to the document. I can use the native one instead (the typescript interface defined in global) if that's allright.

Reply:
> I can use the native one instead (the typescript interface defined in global) if that's allright.

Oh right, I was mistaken in telling you to use the react one.  The native one is the correct one here.